src/unity/python/turicreate/toolkits/image_similarity/image_similarity.py

```diff
@@ -491,40 +507,84 @@ def export_coreml(self, filename):
-        _mxnet_converter.convert(mx_feature_extractor, mode=None,
-                                 input_shape=[(input_name, batch_input_shape)],
-                                 builder=builder, verbose=False)
+        if self.model != 'VisionFeaturePrint_Screen':
+            # Convert the MxNet model to Core ML
+            ptModel = _pre_trained_models.MODELS[self.model]()
+            feature_extractor = _image_feature_extractor.MXFeatureExtractor(ptModel)
+
+            input_name = feature_extractor.data_layer
+            input_features = [(input_name, _datatypes.Array(*(self.input_image_shape)))]
+
+            # Create a neural network
+            builder = _neural_network.NeuralNetworkBuilder(
+                input_features, output_features, mode=None)
+
+            # Convert the feature extraction network
+            mx_feature_extractor = feature_extractor._get_mx_module(
+                feature_extractor.ptModel.mxmodel,
+                feature_extractor.data_layer,
+                feature_extractor.feature_layer,
+                feature_extractor.context,
+                self.input_image_shape
+            )
+            batch_input_shape = (1, ) + self.input_image_shape
+            _mxnet_converter.convert(mx_feature_extractor, mode=None,
+                                     input_shape=[(input_name, batch_input_shape)],
+                                     builder=builder, verbose=False)
+            feature_layer = feature_extractor.feature_layer
+
+        else:     # self.model == VisionFeaturePrint_Screen
+            # Create a pipleline that contains a VisionFeaturePrint followed by a
+            # neural network.
+            BGR_VALUE = _cmt.proto.FeatureTypes_pb2.ImageFeatureType.ColorSpace.Value('BGR')
+            DOUBLE_ARRAY_VALUE = _cmt.proto.FeatureTypes_pb2.ArrayFeatureType.ArrayDataType.Value('DOUBLE')
+
+            top_spec = _cmt.proto.Model_pb2.Model()
+            top_spec.specificationVersion = 3
+            desc = top_spec.description
+
+            input = desc.input.add()
+            input.name = self.feature
+            input.type.imageType.width = 299
+            input.type.imageType.height = 299
+            input.type.imageType.colorSpace = BGR_VALUE
+
+            output = desc.output.add()
+            output.name = output_name
+            output.type.multiArrayType.shape.append(num_examples)
+            output.type.multiArrayType.dataType = DOUBLE_ARRAY_VALUE
+
+            # VisionFeaturePrint extractor
+            pipeline = top_spec.pipeline
+            scene_print = pipeline.models.add()
+            scene_print.specificationVersion = 3
+            scene_print.visionFeaturePrint.scene.version = 1
+
+            input = scene_print.description.input.add()
+            input.name = self.feature
+            input.type.imageType.width = 299
+            input.type.imageType.height = 299
+            input.type.imageType.colorSpace = BGR_VALUE
```

Comment:
Do all image feature extractors in TC use BGR? I'm looking inside `mlmodel_image_feature_extractor.mm` and that seems to be the case. So are all the MXNet models also trained on BGR?

Reply:
I'm not sure. I believe the all the Core ML models used by image classification/similarity use BGR. I believe the MxNet models use RGB.